vending-app/src/main/kotlin/com/google/android/phonesky/header/PhoneskyHeaderValue.kt

```diff
@@ -0,0 +1,275 @@
+                    .string("")
+                    .timestamp(makeTimestamp(System.currentTimeMillis())).build()
+            )
+            .unknown3(0)
+            .build()
+        val encodedLocality = String(
+            Base64.encode(locality.encode(), Base64.URL_SAFE or Base64.NO_WRAP or Base64.NO_PADDING)
+        )
+        val header = LicenseRequestHeader.Builder()
+            .encodedTimestamps(StringWrapper.Builder().string(encodedTimestamps).build())
+            .triple(
+                EncodedTripleWrapper.Builder().triple(
+                    EncodedTriple.Builder()
+                        .encoded1("")
+                        .encoded2("")
+                        .empty("")
+                        .build()
+                ).build()
+            )
+            .locality(LocalityWrapper.Builder().encodedLocalityProto(encodedLocality).build())
+            .unknown(IntWrapper.Builder().integer(5).build())
+            .empty("")
+            .languages(externalxpsrh)
+            .deviceMeta(
+                DeviceMeta.Builder()
+                    .android(
+                        AndroidVersionMeta.Builder()
+                            .androidSdk(org.microg.gms.profile.Build.VERSION.SDK_INT)
+                            .buildNumber(org.microg.gms.profile.Build.ID)
+                            .androidVersion(org.microg.gms.profile.Build.VERSION.RELEASE)
+                            .unknown(0)
+                            .build()
+                    )
+                    .unknown1(UnknownByte12.Builder().bytes(ByteString.EMPTY).build())
+                    .unknown2(1)
+                    .build()
+            )
+            .userAgent(
+                UserAgent.Builder()
+                    .deviceName(org.microg.gms.profile.Build.DEVICE)
+                    .deviceHardware(org.microg.gms.profile.Build.HARDWARE)
+                    .deviceModelName(org.microg.gms.profile.Build.MODEL)
+                    .finskyVersion(FINSKY_VERSION)
+                    .deviceProductName(org.microg.gms.profile.Build.MODEL)
+                    .androidId(androidId) // must not be 0
+                    .buildFingerprint(org.microg.gms.profile.Build.FINGERPRINT)
+                    .build()
+            )
+            .uuid(
+                Uuid.Builder()
+                    .uuid(UUID.randomUUID().toString())
+                    .unknown(2)
+                    .build()
+            )
+            .build().encode()
+        return String(Base64.encode(Util.encodeGzip(header), Base64.URL_SAFE or Base64.NO_WRAP or Base64.NO_PADDING))
+    }
+
+    private fun getHeaders(): Map<String, String> {
+        headerMap["X-PS-RH"] = getXHeaders()
+        headerMap["Authorization"] = "Bearer " + AccountManager.get(context).getAuthToken(
+            user, tokenType, null, false, null, null
+        ).result.getString(AccountManager.KEY_AUTHTOKEN)
+        return this.headerMap
+    }
```

Comment:
This seems to mostly overlap with `getLicenseRequestHeaders` in `LicenseRequestHeaders.kt`. Maybe you can refactor this into a common codebase.